src/index.ts

```diff
@@ -0,0 +1,257 @@
+    // move the 'msg' to 'message'.
+    if (!record.message) {
+      // If this is an error, report the full stack trace. This allows
+      // Stackdriver Error Reporting to pick up errors automatically (for
+      // severity 'error' or higher). In this case we leave the 'msg' property
+      // intact.
+      // https://cloud.google.com/error-reporting/docs/formatting-error-messages
+      //
+      if (record.err && record.err.stack) {
+        record.message = record.err.stack;
+        record.serviceContext = this.serviceContext;
+      } else if (record.msg) {
+        // Simply rename `msg` to `message`.
+        record.message = record.msg;
+        delete record.msg;
+      }
+    }
+
+    const entryMetadata: StackdriverEntryMetadata = {
+      resource: this.resource,
+      timestamp: record.time,
+      // BUNYAN_TO_STACKDRIVER does not have index signature.
+      // tslint:disable-next-line:no-any
+      severity: (BUNYAN_TO_STACKDRIVER as any)[record.level as string],
+    };
+
+    // If the record contains a httpRequest property, provide it on the entry
+    // metadata. This allows Stackdriver to use request log formatting.
+    // https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#HttpRequest
+    // Note that the httpRequest field must properly validate as a HttpRequest
+    // proto message, or the log entry would be rejected by the API. We do no
+    // validation here.
+    if (record.httpRequest) {
+      entryMetadata.httpRequest = record.httpRequest;
+      delete record.httpRequest;
+    }
+
+    // record does not have index signature.
+    // tslint:disable-next-line:no-any
+    if ((record as any)[LOGGING_TRACE_KEY]) {
+      // tslint:disable-next-line:no-any
+      entryMetadata.trace = (record as any)[LOGGING_TRACE_KEY];
+      // tslint:disable-next-line:no-any
+      delete (record as any)[LOGGING_TRACE_KEY];
+    }
+
+    return this.stackdriverLog.entry(entryMetadata, record);
+  }
+
+  /**
+   * Intercept log entries as they are written so we can attempt to add the
+   * trace ID in the same continuation as the function that wrote the log,
+   * because the trace agent currently uses continuation local storage for the
+   * trace context.
+   *
+   * By the time the Writable stream buffer gets flushed and _write gets called
+   * we may well be in a different continuation.
+   */
+  // Writable.write used 'any' in function signature.
+  // tslint:disable-next-line:no-any
+  write(record: any, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(record: any, encoding?: string, callback?: Function): boolean;
+  // tslint:disable-next-line:no-any
+  write(...args: any[]): boolean {
+    let record = args[0];
+    let encoding: string|null = null;
+    let callback: Function;
+    if (typeof args[1] === 'string') {
+      encoding = args[1];
+      callback = args[2];
+    } else {
+      callback = args[1];
+    }
+    record = extend({}, record);
```

Comment:
With `Object.assign` do we need the extend module?

Reply:
Done.